src/Chapter23/Listing23.20.DesignatingABlockForUnsafeCode.cs

```diff
@@ -91,14 +91,20 @@ public override bool IsInvalid
         // SafeHandle abstract member
         protected override bool ReleaseHandle()
         {
-            if (!Disposed)
-            {
-                Disposed = true;
-                GC.SuppressFinalize(this);
-                VirtualMemoryManager.VirtualFreeEx(ProcessHandle,
-                    AllocatedPointer, MemorySize);
-            }
-            return true;
+            return Disposed = VirtualMemoryManager.VirtualFreeEx(ProcessHandle, AllocatedPointer, MemorySize);
+        }
+
+        ~VirtualMemoryPtr()
+        {
+            // Do not change this code. Put cleanup code in 'Dispose(bool disposing)' method
+            Dispose(disposing: false);
+        }
+
+        void IDisposable.Dispose()
+        {
+            // Do not change this code. Put cleanup code in 'Dispose(bool disposing)' method
+            Dispose(disposing: true);
+            GC.SuppressFinalize(this);
```

Comment:
I would propose removing these as well since the SafeHandle already defines them.